packages/core/src/command/rate-limiting/rate-limiting.ts

```diff
@@ -0,0 +1,54 @@
+import { MetriportError, RateLimit, RateLimitOperation } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+import { getCxRateSettingValue } from "./commands/get-settings";
+import { getTrackedOperationCountSum } from "./commands/get-tracked-operation-count-sum";
+import { updateTrackedOperationCount } from "./commands/update-tracked-operation-count";
+import { secondGranularityIsoDateTime, secondsLookup } from "./shared";
+
+export async function checkRateLimit({
+  cxId,
+  operation,
+  rateLimit,
+  client,
+}: {
+  cxId: string;
+  operation: RateLimitOperation;
+  rateLimit: RateLimit;
+  client?: DocumentClient;
+}): Promise<boolean> {
+  const secondsLookback = secondsLookup.get(rateLimit);
+  if (!secondsLookback) {
+    throw new MetriportError("Operation missing seconds lookup", undefined, { operation });
+  }
+
+  const end = buildDayjs();
+  const endStr = end.format(secondGranularityIsoDateTime);
+  const start = end.subtract(secondsLookback, "seconds");
+  const startStr = start.format(secondGranularityIsoDateTime);
+  const [currentCount, limit] = await Promise.all([
+    getTrackedOperationCountSum({
+      cxId,
+      operation,
+      start: startStr,
+      end: endStr,
+      client,
+    }),
+    getCxRateSettingValue({
+      cxId,
+      operation,
+      rateLimit,
+      client,
+    }),
+  ]);
+  if (currentCount === undefined || limit === undefined) return true;
```

Comment:
Aren't we missing updating the tracked operation count when `currentCount` is `undefined`? The ways it is right now, we'll never increase the `numberOfOperation`, right?

Reply:
no longer relevant.